### PR TITLE
fix(smart-action): do not mutate hooks on schema generation

### PR DESCRIPTION
**Pull request by rap2hpoutre**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

### CODE REVIEW

File: src/services/schema-file-updater.js [hunk 1]
```diff
@@ -163,14 +154,33 @@ class SchemaFileUpdater {
...
+      actionFormatted.hooks = actionFormatted.hooks || {};
+      actionFormatted.hooks = SchemaFileUpdater
+        .formatObject(action.hooks, serializerOptions.actions.hooks.attributes);
+      actionFormatted.hooks.load = Boolean(action.hooks && (typeof action.hooks.load === 'function'));
+      actionFormatted.hooks.change = action.hooks
+        && action.hooks.change
+        && typeof action.hooks.change === 'object'
+        ? Object.keys(action.hooks.change)
+        : [];
+
```
Comment (GuillaumeCisco) [inline src/services/schema-file-updater.js:167]: Should we create a method entitled  `SchemaFileUpdater.cleanActionHooks` as we can see for `SchemaFileUpdater.cleanSegments` ?

**Review by GuillaumeCisco — COMMENTED** [review]
Some little comments.
It is hard to follow what really changed.
Can you enlighten me please 🙏

File: src/services/schema-file-updater.js [hunk 2]
```diff
@@ -107,15 +107,7 @@ class SchemaFileUpdater {
...
-  static cleanActionHooks(action) {
-    const load = Boolean(action.hooks && (typeof action.hooks.load === 'function'));
-    const change = action.hooks && action.hooks.change && typeof action.hooks.change === 'object'
-      ? Object.keys(action.hooks.change)
-      : [];
-    action.hooks = { load, change };
-  }
-
-  cleanCollection(collection) {
+  static cleanCollection(collection) {
```
[the remark this answers is not in the document]
Comment (rap2hpoutre) [reply src/services/schema-file-updater.js]: Oops, fixed during my self review

File: src/services/schema-file-updater.js [hunk 3]
```diff
@@ -163,14 +154,33 @@ class SchemaFileUpdater {
...
   }
 
+  static cleanActionHooks(action) {
+    const load = Boolean(action.hooks && (typeof action.hooks.load === 'function'));
+    const change = action.hooks && action.hooks.change && typeof action.hooks.change === 'object'
+      ? Object.keys(action.hooks.change)
+      : [];
+    action.hooks = { load, change };
+  }
+
```
[the remark this answers is not in the document]
Comment (rap2hpoutre) [reply src/services/schema-file-updater.js]: Woops, it was totally broken.

File: src/services/schema-file-updater.js (hunk 1 above)
Comment (rap2hpoutre) [reply src/services/schema-file-updater.js]: Yes! Done.

**Review by GuillaumeCisco — APPROVED** [review]
Great!
Far more easier to read!
🎸 🥁 🎷
